Modix.Data/Models/Core/MessageContentPatternEntity.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Modix.Data.Models.Core
+{
+    [Table("MessageContentPatterns")]
+    public class MessageContentPatternEntity
+    {
+        [Key]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public long Id { get; set; }
+
+        [Required]
+        public string Pattern { get; set; }
+
+        public MessageContentPatternType PatternType { get; set; }
+
+        public ulong GuildId { get; set; }
+    }
+
+    public class MessageContentPatternEntityConfiguration
```

Comment:
Might make sense to have a unique constraint on (GuildId, Pattern) to ensure we can't have bad data